src/components/Form/TimeInput/index.js

```diff
@@ -135,20 +135,35 @@ TimeInput.propTypes = {
   rules: PropTypes.shape({}),
 }
 
-const ControlledTimeInput = ({ control, name, rules, ...otherProps }) => (
-  <Controller
-    as={<TimeInput {...otherProps} />}
-    onChange={([value]) => value}
-    name={name}
-    rules={{
-      ...rules,
-      validate: {
-        valid: (value) => {
-          if (value) {
-            let hourStr = value.split(':')[0]
-            let minuteStr = value.split(':')[1]
-            let hour = parseInt(hourStr)
-            let minute = parseInt(minuteStr)
+const ControlledTimeInput = ({
+  control,
+  name,
+  label,
+  emptyErrorMessage,
+  required,
+  ...otherProps
+}) => {
+  return (
+    <Controller
+      as={<TimeInput label={label} {...otherProps} />}
+      onChange={([value]) => value}
+      rules={{
+        validate: {
+          valid: (value) => {
+            // required - validate being empty
+            if (required && !value) return emptyErrorMessage
+
+            // not required - valid if empty
+            if (!required && !value) return true
+
+            const hourStr = value.split(':')[0] || ''
+            const minuteStr = value.split(':')[1] || ''
+
+            const hour = parseInt(hourStr)
+            const minute = parseInt(minuteStr)
+
+            if (isNaN(hour) || isNaN(minute)) return 'Please enter a valid time'
```

Comment:
String is duplicated with line 178. Also, if emptyErrorMessage is a prop, why isn't the invalidErrorMessage? Happy with either approach, but we are either promoting this component for re-useability or we are not

Reply:
I was trying to use the label in the error message but had issues with '(optional)' being part of the label. 

I have replaced the first error with the following:
```javascript
if (required && !value) return `Please enter a ${label.toLowerCase()}`
```

Then added a `showAsOptional` parameter to the component.